src/rules/object-properties.ts

```diff
@@ -23,28 +24,95 @@ function groupNodes(properties: (Property | SpreadElement)[]) {
   return groups.filter((group) => group.length > 1)
 }
 
+/**
+ * Functions which check that the given 2 names are in specific order.
+ *
+ * Postfix `I` is meant insensitive.
+ * Postfix `N` is meant natural.
+ *
+ * Adapted from Eslint's sort-keys rule
+ */
+const sorters: { [key in string]: (a: string, b: string) => number } = {
+  asc(a, b) {
+    return a.localeCompare(b)
+  },
+  ascI(a, b) {
+    return a.toLowerCase().localeCompare(b.toLowerCase())
+  },
+  ascN(a, b) {
+    return naturalCompare(a, b)
+  },
+  ascIN(a, b) {
+    return naturalCompare(a.toLowerCase(), b.toLowerCase())
+  },
+  desc(a, b) {
+    return sorters.asc(b, a)
+  },
+  descI(a, b) {
+    return sorters.ascI(b, a)
+  },
+  descN(a, b) {
+    return sorters.ascN(b, a)
+  },
+  descIN(a, b) {
+    return sorters.ascIN(b, a)
+  },
+}
+
 export default {
   create(context) {
+    const order = context.options[0] || "asc"
+    const options = context.options[1]
+
+    const isCaseSensitive = options?.caseSensitive
+    const isNaturalOrder = options?.natural
+
+    const sorter =
+      sorters[
+        `${order}${isCaseSensitive ? "" : "I"}${isNaturalOrder ? "N" : ""}`
+      ]
+
     return {
       ObjectExpression(expression) {
         for (const nodes of groupNodes(expression.properties)) {
           const sorted = nodes
             .slice()
-            .sort(alphaSorter((node) => getName(node.key).toLowerCase()))
+            .sort((nodeA, nodeB) =>
+              sorter(getName(nodeA.key), getName(nodeB.key))
+            )
 
           report(context, nodes, sorted)
         }
       },
     }
   },
   meta: {
-    type: "suggestion",
     docs: {
       url: docsURL("object-properties"),
     },
     fixable: "code",
     messages: {
       unsorted: "Object properties should be sorted alphabetically.",
     },
+    schema: [
+      {
+        enum: ["asc", "desc"],
```

Comment:
Is there a reason for supporting descending? I haven't added it before since it seems so unnecessary and just is extra maintenance burden.

Reply:
@mskelton yeah good point, probably not. I added it because it's in line with `sort-key`, but I agree that it doesn't really add any value